raiden/network/transport/matrix/rtc/web_rtc.py

```diff
@@ -0,0 +1,229 @@
+
+    return gevent.spawn(
+        wait_for_future, asyncio.ensure_future(coroutine), callback, partner_address
+    )
+
+
+def wait_for_future(future: Any, callback: Callable, partner_address: Address) -> None:
+    result = yield_future(future)
+    if callback is not None:
+        callback(result, partner_address)
+
+
+class WebRTCManager:
+    def __init__(
+        self,
+        node_address: Optional[Address],
+        handle_message_callback: Callable[[str, Address], None],
+        handle_sdp_callback: Callable[[Optional[RTCSessionDescription], Address], None],
+    ) -> None:
+        self.node_address: Optional[Address] = node_address
+        self._handle_message_callback = handle_message_callback
+        self._handle_sdp_callback = handle_sdp_callback
+        self.address_to_rtc_partners: Dict[Address, RTCPartner] = dict()
+        self.coroutines: List[Coroutine] = list()
+
+    def get_rtc_partner(self, partner_address: Address) -> RTCPartner:
+        if partner_address not in self.address_to_rtc_partners:
+            self.address_to_rtc_partners[partner_address] = RTCPartner(
+                partner_address, RTCPeerConnection()
+            )
+        return self.address_to_rtc_partners[partner_address]
+
+    def has_ready_channel(self, partner_address: Address) -> bool:
+        if partner_address not in self.address_to_rtc_partners:
+            return False
+        channel = self.address_to_rtc_partners[partner_address].channel
+        if channel is None:
+            return False
+        if channel.readyState == "open":
+            return True
+        return False
+
+    def async_create_channel(self, partner_address: Address) -> None:
+        assert self.node_address, "Transport is not started yet but tried to create rtc channel"
+
+        rtc_partner = self.get_rtc_partner(partner_address)
+        spawn_coroutine(
+            coroutine=create_channel(
+                rtc_partner, self.node_address, self._handle_message_callback
+            ),
+            callback=self._handle_sdp_callback,
+            partner_address=partner_address,
+        )
+
+    def async_set_remote_description(
+        self, partner_address: Address, description: Dict[str, str]
+    ) -> None:
+        assert self.node_address, "Transport is not started yet but tried to set candidates"
+        rtc_partner = self.get_rtc_partner(partner_address)
+        spawn_coroutine(
+            coroutine=set_remote_description(
+                rtc_partner=rtc_partner,
+                node_address=self.node_address,
+                description=description,
+                handle_message_callback=self._handle_message_callback,
+            ),
+            callback=self._handle_sdp_callback,
+            partner_address=partner_address,
+        )
+
+    def stop(self) -> None:
+        if self.node_address is None:
+            return
+
+        log.debug("Gracefully closing RTC channels", node=to_checksum_address(self.node_address))
```

Comment:
In some logs `rtc` is used, in other `RTC`. Can you use a consistent spelling?

Reply:
done